modules/videoio/src/cap_gstreamer.cpp

```diff
@@ -208,6 +208,16 @@ void CvCapture_GStreamer::close()
         pipeline = NULL;
     }
 
+    if(buffer) {
+        gst_buffer_unref(buffer);
+        buffer = NULL;
+    }
```

Comment:
this is wrong for GStreamer 1.x versions. See https://developer.gnome.org/gstreamer/stable/gstreamer-GstSample.html. I suppose you have to unref `sample` object for GStreamer 1.x